controllers/ocirepository_controller.go

```diff
@@ -408,6 +411,24 @@ func (r *OCIRepositoryReconciler) reconcileSource(ctx context.Context, obj *sour
 
 	// Extract the content of the first artifact layer
 	if !obj.GetArtifact().HasRevision(revision) {
+		if obj.Spec.Verify != nil {
```

Comment:
The verification being only when there's a revision mismatch results in no verification when there's an existing artifact and verification is enabled later without update to the image.
In case of GitRepository, verification is done irrespective of artifact revision mismatch, refer https://github.com/fluxcd/source-controller/blob/54d706a226e04426adb8e25de8b0fd03e7375360/controllers/gitrepository_controller.go#L533-L539 .

Reply:
Unlike the OpenPGP verification which relies on the in-cluster objects, the Cosign verification is doing lots of external calls to the container registry plus and to some Google APIs plus Rekor etc. If we call verify every time, we'll end up hitting the rate limits and increase the outbound network costs.

Reply:
Found a way to detect if verify was added after the artifact was downloaded.